tests/test_models.cc

```diff
@@ -255,4 +255,54 @@ TEST(test_models, test_model_from_prediction_low_rank) {
       model_pred.covariance, 1e-8));
 }
 
+TEST(test_models, test_model_from_different_datasets) {
+  Eigen::Index k = 10;
+  Eigen::VectorXd mean = 3.14159 * Eigen::VectorXd::Ones(k);
+  Eigen::VectorXd variance = 0.1 * Eigen::VectorXd::Ones(k);
+  MarginalDistribution targets(mean, variance.asDiagonal());
+
+  std::vector<double> train_features;
+  for (Eigen::Index i = 0; i < k; ++i) {
+    train_features.push_back(static_cast<double>(i) * 0.3);
+  }
+
+  ConstantEverywhere constant;
+  ConstantPerInterval per_interval;
+
+  // First we fit a model directly to the training data and use
+  // that to get a prediction of the inducing points.
+  auto model = gp_from_covariance(constant + per_interval, "unobservable");
+  const auto fit_model = model.fit(train_features, targets);
+  const auto inducing_points = create_inducing_points(train_features);
```

Comment:
This test really hinges on what this function does and whether `InducingFeature` is or is not a double. I cannot tell that from the test alone, but assume that it's testing the fit using two different datasets.

Reply:
`InducingFeature` is defined earlier in the `.cc` file (but wasn't changed so you probably can't see it.).

```
enum InducingFeatureType { ConstantEverywhereType, ConstantPerIntervalType };

struct InducingFeature {
  InducingFeatureType type;
  long location;
};
enum InducingFeatureType { ConstantEverywhereType, ConstantPerIntervalType };

struct InducingFeature {
  InducingFeatureType type;
  long location;
};
```
And yeah, what it's testing is that we can fit a model with two datasets which are defined for two different types .